client/app/lib/redux/modules/sidebar/stacks.coffee

```diff
@@ -0,0 +1,306 @@
+_ = require 'lodash'
+kd = require 'kd'
+isAdmin = require 'app/util/isAdmin'
+immutable = require 'app/util/immutable'
+showError = require 'app/util/showError'
+whoami = require 'app/util/whoami'
+getMachineOwner = require 'app/util/getMachineOwner'
+isStackTemplateSharedWithTeam = require 'app/util/isstacktemplatesharedwithteam'
+{ series, waterfall } = require 'async'
+{ createSelector } = require 'reselect'
+
+
+{ makeNamespace, expandActionType,
+  normalize, defineSchema } = require 'app/redux/helper'
+
+withNamespace = makeNamespace 'koding', 'sidebar', 'stacks'
+
+GENERATE_STACK = expandActionType withNamespace 'GENERATE_STACK'
+EDIT = expandActionType withNamespace 'EDIT'
+INITIALIZE = expandActionType withNamespace 'INITIALIZE'
+{ LOAD, REMOVE } = require 'app/redux/modules/bongo'
+
+bongo = require 'app/redux/modules/bongo'
+
+reducer = (state = immutable({}), action) ->
+
+  switch action.type
+
+    when 'STACK_REVISION_SUCCESS'
+      { result: { _id, data, error } } = action
+      return state  if error
+      state = state.set _id, immutable data
+
+      return state
+
+    else
+      return state
+
+# actions
+
+initializeStack = (template) -> (dispatch) ->
+  { reactor } = kd.singletons
+
+  waterfall([
+    (next) ->
+      dispatch {
+        types: [LOAD.BEGIN, LOAD.SUCCESS, LOAD.FAIL]
+        bongo: -> template.generateStack().then (result) ->
+          { stack, results: { machines } } = result
+          instances = []
+          instances.push stack
+          machines.forEach (machine) -> instances.push machine.obj
+          next null, machines[0].obj
+          reactor.dispatch 'GENERATE_STACK_SUCCESS', { template, stack }
+
+          return instances
+        }
+
+    (machine, next) ->
+      console.log '???? ', machine, ' should be null amk', machine.label
+      new kd.NotificationView { title: 'Stack generated successfully' }
+      reloadIDE machine.label
+      next()
+  ])
+  console.log 'something'
+
+
+openOnGitlab = (stack) ->
```

Comment:
This can be a key for `mapDispatchToProps`. This doesn't have to do anything with this module. Instead move this function into the container component.